packages/opentelemetry-sampler-aws-xray/src/sampling-rule.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ISamplingRule } from './types';
+import { wildcardMatch, attributeMatch } from './utils';
+import { SamplingDecision } from '@opentelemetry/sdk-trace-base';
+import { Attributes, AttributeValue } from '@opentelemetry/api';
+import {
+  SemanticAttributes,
+  SemanticResourceAttributes,
+} from '@opentelemetry/semantic-conventions';
+import { Resource } from '@opentelemetry/resources';
+import { Reservoir } from './reservoir';
+import { Statistics } from './statistics';
+
+export class SamplingRule implements ISamplingRule {
+  public RuleName: string;
+  public Priority: number;
+  public ReservoirSize: number;
+  public FixedRate: number;
+  public ServiceName: string;
+  public ServiceType: string;
+  public Host: string;
+  public HTTPMethod: string;
+  public URLPath: string;
+  public ResourceARN: string;
+  public Attributes?: { [key: string]: string };
+  public Version: number;
+  public reservoir: Reservoir;
+  public statistics: Statistics;
+
+  constructor(samplingRule: any) {
+    this.RuleName = samplingRule.ruleName;
+    this.Priority = samplingRule.priority;
+    this.ReservoirSize = samplingRule.reservoirSize;
+    this.FixedRate = samplingRule.fixedRate;
+    this.ServiceName = samplingRule.serviceName;
+    this.ServiceType = samplingRule.serviceType;
+    this.Host = samplingRule.host;
+    this.HTTPMethod = samplingRule.httpMethod;
+    this.URLPath = samplingRule.urlPath;
+    this.ResourceARN = samplingRule.resourceARN;
+    this.Version = samplingRule.version;
+    this.Attributes = samplingRule.attributes;
```

Comment:
Instead of copying the fields from one object to another, we can alternatively just store `samplingRule` as a property on the class and access its values, no?
Or is there anything I am missing?

Reply:
Sure, this is how we have been implementing the sampling rule class across other OTEL languages so this format was following the other languages for consistency